extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/PolicyEnforcerResolver.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.keycloak.pep;
+
+import org.keycloak.adapters.authorization.PolicyEnforcer;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A {@link PolicyEnforcer} resolver.
+ */
+public interface PolicyEnforcerResolver {
+
+    Uni<PolicyEnforcer> resolvePolicyEnforcer(RoutingContext routingContext, String tenantId);
+
+    long getReadTimeout();
```

Comment:
Why this method should be here ?

Reply:
It is tenant specific https://quarkus.io/guides/all-config#quarkus-keycloak-authorization_quarkus-keycloak-tenant-connection-pool-size and if someone decides to define it's own `PolicyEnforcerResolver` than this timeout is used when `resolvePolicyEnforcer` is invoked, so you might need longer read timeout for certain tenants. That is in the configuration already.

We provide our `PolicyEnforcerResolver` for users and while in CDI world we can't stop users from defining their own, like #25774 asked for, users shouldn't have reasons to do that.

Reply:
Sorry, wrong link, let me update the comment in the minute.